Adds WatchOS target to the app

nealsanche opened the pull request:
### Introduction

I've been wanting something like this for quite a while, and had one of the guys at work give me a hand when I couldn't figure out the final details that I needed to get this working. Turned out, the trick here was that in the Podfile I needed to specify the WatchKit Extension as the target, and that got things to run.

This could probably stand some cleanup. Also I noticed that the app wouldn't run on the HEAD of your code, maybe because of the changes to have native coroutines. WatchOS support may be lacking this? Or perhaps it's a dependency issue. Could maybe use your eyes on that.

Thanks again for the simple example, it really helps. The documentation that's out there is sparse.

Also, I don't really expect this to be merged into your repo directly. When I look at the code that was generated by the XCode IDE, I can see my name all over it. I'll go though an comment on the main points where I had to make changes.

nealsanche commented on watchos/PeopleInSpaceWatch/PeopleInSpaceWatch WatchKit Extension/ContentView.swift:
```diff
@@ -0,0 +1,46 @@
+            List(peopleInSpaceViewModel.people, id: \.name) { person in
+                PersonView(person: person)
+            }
+            //.navigationBarTitle(Text("PeopleInSpace"), displayMode: .large)
```
This needs a bit of work. It turns out if you use a NavigationLink inside PersonView below, you can get some master detail view navigation for free. I'll add that in for the example.

nealsanche commented on common/build.gradle:
```diff
@@ -40,18 +40,24 @@ kotlin {
 
-        fromPreset(presets.android, 'android')
-    }
+        final def watchTarget = System.getenv('SDK_NAME')?.startsWith("watchos") \
```
This section creates the watch target.

joreilly commented:
Regarding native multithreaded coroutine support did you try following for `watchMain` as well?

```
            implementation('org.jetbrains.kotlinx:kotlinx-coroutines-core-native') {
                version {
                    strictly '1.3.3-native-mt'
                }
            }
```

nealsanche commented on common/build.gradle:
```diff
@@ -104,5 +110,19 @@ kotlin {
             implementation "org.jetbrains.kotlinx:kotlinx-serialization-runtime-native:${Versions.kotlinxSerialization}"
         }
 
+        watchMain.dependencies {
```
Here's the watch dependencies.

nealsanche commented on common/common.podspec:
```diff
@@ -17,6 +17,10 @@ Pod::Spec.new do |spec|
     spec.pod_target_xcconfig = {
         'KOTLIN_TARGET[sdk=iphonesimulator*]' => 'ios_x64',
         'KOTLIN_TARGET[sdk=iphoneos*]' => 'ios_arm',
+        'KOTLIN_TARGET[sdk=watchsimulator*]' => 'watchos_x86',
```
These changes were the result of running the Gradle task to generate the podspec.

nealsanche commented on common/src/watchMain/kotlin/com/surrus/common/Dispatcher.kt:
```diff
@@ -0,0 +1,19 @@
+import platform.darwin.*
+
+
+internal actual val ApplicationDispatcher: CoroutineDispatcher =
```
Added this actual which is the same as the iOS actual.

joreilly replied:
With addition of the native coroutines dependency these (`Dispatcher.kt`) were no longer needed.  I guess fork was probably created before that change was made?

nealsanche commented on watchos/PeopleInSpaceWatch/Podfile:
```diff
@@ -0,0 +1,4 @@
+target 'PeopleInSpaceWatch WatchKit Extension' do
```
This was the line of the Podfile that took me the most time to figure out. You have to set the target correctly otherwise the common framework isn't found in the UI code.

nealsanche commented:
> Regarding native multithreaded coroutine support did you try following for `watchMain` as well?
> 
> ```
>             implementation('org.jetbrains.kotlinx:kotlinx-coroutines-core-native') {
>                 version {
>                     strictly '1.3.3-native-mt'
>                 }
>             }
> ```

No, I didn't try that. Which would explain why it didn't run. :) That's probably the bit I missed.

nealsanche commented:
Unfortunately, I left my Mac at work, so I'll work on rebasing this onto the HEAD of your branch, update the coroutines handling, get rid of the notification scene for the watch app, and so on.

nealsanche commented:
Okay, I've rebased on the head of master, and updated the dependencies. Also updated the bundle id of the watch app to be consistent (using your domain). I also removed any code headers that mention me. I removed the notification scene and for fun added a NavigationLink in the Person view for the watch, which just displays the craft the person is on. Makes the app a tiny bit interactive.

joreilly commented:
Looks good, thanks.  I'll add something in README about this after it's merged
